containerized_ml_models/model_api_2/api.py

```diff
@@ -0,0 +1,32 @@
+from flask import Flask, jsonify, request
+from pickle import load as pkl_load
+from ast import literal_eval
+from numpy import asarray
+
+app = Flask(__name__)
+MODEL_NAME = "model.pkl"
+
+
+def load_pickle(fileName):
+    return pkl_load(open(MODEL_NAME, "rb"))
+
+
+@app.route("/ping", methods=["GET"])
+def ping():
+    return jsonify({"ping": "success"})
+
+
+@app.route("/predict", methods=["POST"])
+def predict():
+    model = load_pickle()
+    data = request.get_data(as_text=True)
+
+    input = literal_eval(data)
+    input = asarray(input)
+    output = model.predict(input)
+
+    return jsonify({"Result": f"{output}"})
+
+
+if __name__ == "__main__":
+    app.run()
```

Comment:
Watch this. I had the same problem after this video managed to run on docker
https://www.youtube.com/watch?v=4uoWRXuYfJs

Reply:
Look at the Dockerfile. I'm running the api with "flask --app ...", doing so I'm able to pass the hostname and ports in the Dockerfile's entrypoint.